tests/trainer/test_trainer.py

```diff
@@ -1624,6 +1624,50 @@ def test_torchdynamo_full_eval(self):
         metrics = trainer.evaluate()
         self.assertAlmostEqual(metrics["eval_loss"], original_eval_loss)
 
+    @require_torch_gpu
+    @require_torchdynamo
+    def test_torchdynamo_memory(self):
+        class MyModule(torch.nn.Module):
+            """Simple module that does aggressive fusion"""
+
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                for _ in range(10):
+                    x = torch.sin(x)
+                return x
+
+        mod = MyModule()
+        a = torch.randn(1024, 1024, device="cuda", requires_grad=True)
+
+        # 1. Default - without TorchDynamo
+        a.grad = None
+        trainer = Trainer(model=mod)
+        peak_mem_at_start = torch.cuda.max_memory_allocated()
```

Comment:
looking good, but don't you need to reset the max counter before each cycle?

Reply:
what about speed, which is the main benefit of this feature?

I have already suggested a template test to copy from which already gives you the stats for memory and speed.

Reply:
I targeted memory because of two reasons

* Memory is little bit more predictable to measure than performance.
* Memory reduction (like this test checks for) also points that there is fusion happening. And fusion would mean performance speedup as well.

Regarding the reseting point you mentioned, I am not doing a whole training. I am just doing a training step here. So, there is no need of resetting the counter after every iteration.

Do you still think that performance speedup is required to be tested here?

Reply:
if you feel that detecting memory shrinking is sufficient to tell that torchdynamo works - that works for me.

> Regarding the reseting point you mentioned, I am not doing a whole training. I am just doing a training step here. So, there is no need of resetting the counter after every iteration.

what does this have to do with whole training vs single step?

If one wants a reliable measurement of peak memory one has to reset it before every measurement. Otherwise the 2nd measurement could be below the first one as it'd report the invalid first experiment's value.

Unless I'm missing something of course.

Reply:
My bad. You are right! I must reset the peak_memory stats before running TorchDynamo. I have added that code now.

Reply:
And yes, memory footprint reduction is sufficient to test this integration.